### ISSUE TITLE
Game bundles

### ISSUE THREAD

**Issue by specing** [springlobby/springlobby#818]
Ideas:
- preconfigure springlobby to be more game-specific (i.e. filter all games but the configured one out, at least until the newbie has grasped the reduced complexity)
- download game bundle at start to offer non-empty singleplayer
- offer a special download menu that downloads game packs

game bundle := game + engine + a few maps [+ a few AIs]

```
{
   version = 1,
   game = "Spring: 1944",
   game_version = "3.00",
   engine = "SpringRTS",
   engine_version  = "103.0",
   maps = [
       {"1944_village_crossing", "v3"),
       {"1944_Titan", ""}
    ],
   ai = null,
   filter = same format as in sl.conf
}
```

TODO: how would game devs "package" SL and/or customise it without having to setup separate build infra? Make existing bots build for all interested games? Put empty array into executables to be pre-filled?

It would be reasonable to do this before #817 as the format doesen't hang in the air and implementing it would not duplicate code for singleplayer&hosting.

**Comment by abma** [comment]
> TODO: how would game devs "package" SL and/or customise it without having to setup separate build infra? Make existing bots build for all interested games?

imo nothing should be recompiled: only repacking of the archives should be needed. i.e. place bundle.json next to springlobby.exe makes springlobby automaticly do some stuff at first start.

**Comment by abma** [comment]
this could be also just a shortcut which starts springlobby with some parameter which makes it download the current version of the bundle file.

**Comment by specing** [comment]
Right, if argv[0] not equal to "springlobby", then try downloading argv[0] as bundle name. Added filename to top comment.

symlink `/bundles/argv0/s44lobby to bundles/Spring1944/Spring1944-3.00.json`? How to do this on the server @abma? I was planning for every team to have push access to their bundle/game_name, but updating lobby like this would require changes in argv0. Perhaps first manually register a lobby name by linking it to bundles/game-name/lobby-name.json?

**Comment by specing** [comment]
Actually this file should have the same configuration tree as springlobby.conf, which should be converted to json. Then this file would override springlobby.conf (hook into cfg()...).
